apollo-router/src/request.rs

```diff
@@ -46,7 +40,23 @@ where
     <Option<T>>::deserialize(deserializer).map(|x| x.unwrap_or_default())
 }
 
+#[buildstructor::buildstructor]
 impl Request {
+    #[builder]
+    pub fn new(
+        query: Option<String>,
+        operation_name: Option<String>,
+        variables: Option<Object>,
+        extensions: Option<Object>,
+    ) -> Self {
+        Self {
+            query,
+            operation_name,
+            variables: Arc::new(variables.unwrap_or_default()),
```

Comment:
Hmm, that seems suspect. Shouldn't we be passing:
```
variables: Arc<Object>
```
? i.e.: it's not optional in the current interface and I'm wondering why it is in the new interface.

Or, if we want to create the `Arc` in the constructor, then:
```
variables: Object
```

Reply:
I think it was optional before:

```rust
    #[serde(
        skip_serializing_if = "Object::is_empty",
        default,
        deserialize_with = "deserialize_null_default"
    )]
    #[builder(default)]
    pub variables: Arc<Object>,
```
    
`#[builder(default)]` was typed builder's way of making this optional, also deserializing from json also gives a default object.